test/e2e/node_shutdown_startup.go

```diff
@@ -0,0 +1,259 @@
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2enode "k8s.io/kubernetes/test/e2e/framework/node"
+	e2eskipper "k8s.io/kubernetes/test/e2e/framework/skipper"
+)
+
+var _ = ginkgo.Describe("Node Shutdown and Startup", ginkgo.Serial, func() {
+	const (
+		nodeShutdownTimeout = 5 * time.Minute
+		nodeStartupTimeout  = 10 * time.Minute
+	)
+
+	var (
+		f            *framework.Framework
+		testNodeName string
+	)
+
+	f = wrappedTestFramework("node-shutdown-startup")
+
+	ginkgo.BeforeEach(func() {
+		// Skip test if not using kind provider
+		if infraprovider.Get().Name() != "kind" {
+			e2eskipper.Skipf("Node shutdown/startup test only supported for kind provider, got: %s", infraprovider.Get().Name())
+		}
+
+		// Get a worker node for testing (skip master/control-plane nodes)
+		nodes, err := e2enode.GetReadySchedulableNodes(context.TODO(), f.ClientSet)
+		framework.ExpectNoError(err, "Failed to get ready schedulable nodes")
+
+		if len(nodes.Items) < 2 {
+			e2eskipper.Skipf("Test requires at least 2 nodes, found %d", len(nodes.Items))
+		}
+
+		// Find a worker node (not master/control-plane)
+		for _, node := range nodes.Items {
+			if !isControlPlaneNode(node) {
+				testNodeName = node.Name
+				break
+			}
+		}
+
+		if testNodeName == "" {
+			e2eskipper.Skipf("No worker nodes found for testing")
+		}
+
+		framework.Logf("Using node %s for shutdown/startup test", testNodeName)
+	})
+
+	ginkgo.It("should maintain cluster health after node shutdown and startup", func() {
+		err := waitOVNKubernetesHealthy(f)
+		framework.ExpectNoError(err, "OVN-Kubernetes cluster should be healthy initially")
+
+		ginkgo.By("Check breth0 IP address families before shutdown")
+		initialIPFamilies, err := getBridgeIPAddressFamilies(testNodeName)
+		framework.ExpectNoError(err, "Should be able to get breth0 IP address families before shutdown")
+		framework.Logf("Node %s breth0 initial IP families: IPv4=%v, IPv6=%v", testNodeName, initialIPFamilies.hasIPv4, initialIPFamilies.hasIPv6)
+
+		ginkgo.By("Shut down the node")
+		framework.Logf("Shutting down node %s", testNodeName)
+		err = infraprovider.Get().ShutdownNode(testNodeName)
+		framework.ExpectNoError(err, "Failed to shutdown node %s", testNodeName)
+
+		// Ensure node is started back up regardless of test failure
+		defer func() {
+			// If the test failed, dump container logs from the node before cleanup
+			if ginkgo.CurrentSpecReport().Failed() {
+				framework.Logf("Test failed, dumping container logs from node %s", testNodeName)
+				dumpContainerLogsFromNode(testNodeName)
+			}
+
+			framework.Logf("Ensuring node %s is started (cleanup)", testNodeName)
+			if startErr := infraprovider.Get().StartNode(testNodeName); startErr != nil {
+				framework.Logf("Failed to start node %s during cleanup: %v", testNodeName, startErr)
+			}
+		}()
+
+		// Wait for the node to be marked as NotReady
+		ginkgo.By("Waiting for node to be marked as NotReady")
```

Comment:
Can we put this node NotReady/Ready status checking to a function that others can reuse it as well?